integration/network/macvlan_test.go

```diff
@@ -48,292 +49,275 @@ func TestDockerNetworkMacvlanPersistance(t *testing.T) {
-	skip.If(t, testEnv.IsRemoteDaemon())
-	skip.If(t, !macvlanKernelSupport(), "Kernel doesn't support macvlan")
-
-	d := daemon.New(t, "", "dockerd", daemon.Config{})
-	d.StartWithBusybox(t)
-	defer d.Stop(t)
-	client, err := d.NewClient()
-	assert.NilError(t, err)
-	_, err = client.NetworkCreate(context.Background(), "dualstackbridge", types.NetworkCreate{
-		Driver:     "macvlan",
-		EnableIPv6: true,
-		Options: map[string]string{
-			"macvlan_mode": "bridge",
-		},
-		IPAM: &network.IPAM{
-			Config: []network.IPAMConfig{
-				{
-					Subnet:     "172.28.130.0/24",
-					AuxAddress: map[string]string{},
-				},
-				{
-					Subnet:     "2001:db8:abca::/64",
-					Gateway:    "2001:db8:abca::254",
-					AuxAddress: map[string]string{},
+func testMacvlanAddressing(client dclient.APIClient) func(*testing.T) {
+	return func(t *testing.T) {
+		// Ensure the default gateways, next-hops and default dev devices are properly set
+		_, err := client.NetworkCreate(context.Background(), "dualstackbridge", types.NetworkCreate{
+			Driver:     "macvlan",
+			EnableIPv6: true,
+			Options: map[string]string{
+				"macvlan_mode": "bridge",
+			},
+			IPAM: &network.IPAM{
+				Config: []network.IPAMConfig{
+					{
+						Subnet:     "172.28.130.0/24",
+						AuxAddress: map[string]string{},
+					},
+					{
+						Subnet:     "2001:db8:abca::/64",
+						Gateway:    "2001:db8:abca::254",
+						AuxAddress: map[string]string{},
+					},
 				},
 			},
-		},
-	})
-	assert.NilError(t, err)
-	assert.Check(t, isNetworkAvailable(client, "dualstackbridge"))
-
-	ctx := context.Background()
-	id1 := container.Run(t, ctx, client,
-		container.WithNetworkMode("dualstackbridge"),
-		container.WithName(t.Name()+"first"),
-	)
-
-	// Validate macvlan bridge mode defaults gateway sets the default IPAM next-hop inferred from the subnet
-	result, err := container.Exec(ctx, client, id1, []string{"ip", "route"})
-	assert.NilError(t, err)
-	assert.Check(t, strings.Contains(result.Combined(), "default via 172.28.130.1 dev eth0"))
-	// Validate macvlan bridge mode sets the v6 gateway to the user specified default gateway/next-hop
-	result, err = container.Exec(ctx, client, id1, []string{"ip", "-6", "route"})
-	assert.NilError(t, err)
-	assert.Check(t, strings.Contains(result.Combined(), "default via 2001:db8:abca::254 dev eth0"))
+		})
+		assert.NilError(t, err)
+		assert.Check(t, isNetworkAvailable(client, "dualstackbridge"))
+
+		ctx := context.Background()
+		id1 := container.Run(t, ctx, client,
+			container.WithNetworkMode("dualstackbridge"),
+		)
+
+		// Validate macvlan bridge mode defaults gateway sets the default IPAM next-hop inferred from the subnet
+		result, err := container.Exec(ctx, client, id1, []string{"ip", "route"})
+		assert.NilError(t, err)
+		assert.Check(t, strings.Contains(result.Combined(), "default via 172.28.130.1 dev eth0"))
+		// Validate macvlan bridge mode sets the v6 gateway to the user specified default gateway/next-hop
```

Comment:
Wondering if we should start moving comments like this as custom message (not for this PR)